stdlib/public/core/OutputStream.swift

```diff
@@ -330,11 +330,14 @@ internal struct _Stdout : OutputStreamType {
   }
 
   mutating func write(string: String) {
-    // FIXME: buffering?
-    // It is important that we use stdio routines in order to correctly
-    // interoperate with stdio buffering.
-    for c in string.utf8 {
-      _swift_stdlib_putchar_unlocked(Int32(c))
+    let utf8 = string.nulTerminatedUTF8
```

Comment:
If the string contains non-ASCII characters, then we don't have the bytes in UTF-8 anywhere in the memory.  They are either in UTF-16 if String is in the Swift representation, or in UTF-16 NSString representation, or in an unknown NSString representation if we are not lucky.